src/actions/expressions.js

```diff
@@ -28,10 +29,21 @@ import type { ThunkArgs } from "./types";
  */
 export function addExpression(input: string) {
   return async ({ dispatch, getState }: ThunkArgs) => {
-    if (!input) {
+    // clear previous error.
+    const prevErr = getExpressionError(getState());
+    const error = await parser.hasSyntaxError(input);
+    if (prevErr && !error) {
+      dispatch(setExpressionError(false));
+    }
```

Comment:
two things seem off here:

1. a clearExpressionError action could consolidate the logic
2. we could clear the error field in the `ADD_EXPRESSION` reducer 

```js
      case "ADD_EXPRESSION":
        return appendToList(state, ["expressions"], {
          input: action.input,
          value: null,
          updating: true
        })
          .set("error", ???)
+     case "EXPRESSION_ERROR":

```

Reply:
Huh I didn't realize I could do it right in the `add/update` cases. I agree on the clear expression action, it's a little clearer to have a distinct function over a flag.